### PR TITLE
DEV: lint: add option to lint all files

### PR DESCRIPTION
**Pull request by lucascolley**
This will be useful to people working on PRs like https://github.com/scipy/scipy/pull/21987

### CODE REVIEW

File: dev.py [hunk 1]
```diff
@@ -1174,26 +1176,30 @@ def task_check_unicode():
...
 
     @classmethod
-    def run(cls, fix):
+    def run(cls, fix, all):
         run_doit_task({
-            'lint': {'fix': fix},
-            'check-unicode': {},
-            'check-testname': {},
+            'lint': {'fix': fix, 'all': all},
+            'check_unicode': {},
```
Comment (lucascolley) [inline dev.py:1201]: this fixes a small bug from gh-22003

File: tools/lint.py [hunk 2]
```diff
@@ -118,9 +145,30 @@ def main():
     parser.add_argument("--files", nargs='*',
                         help="Lint these files or directories; "
                              "use **/*.py to lint all files")
+    parser.add_argument("--all", action='store_true',
+                        help="This overrides `--diff-against` and `--files` "
+                             "to lint all local files (excluding subprojects).")
+    parser.add_argument("--no-cython", dest='cython', action='store_false',
+                       help="Do not run cython-lint.")
```
Comment (lucascolley) [inline tools/lint.py:152]: not sure yet whether/how to expose this via `dev.py`, but perhaps we can punt on that until the `spin` PR is in

File: dev.py (hunk 1 above)
Comment (dschult) [reply dev.py:1201]: Can we make all of these use dash or all use underscore?
Changing `check-unicode` to `unicode_check` when they are sitting next to `check-testname` and `check_python_h_first` is confusing. :}
Not a big deal... just would help me remember better...

Comment (lucascolley) [reply dev.py:1201]: Sorry, I'm not sure what you mean? I think everything changed here in that regard is just implementation details?

Comment (dschult) [reply dev.py:1201]: I see.... people can't do `dev.py check_unicode`.  I was thinking they were developer facing.
Sorry for the noise.

Comment (lucascolley) [reply dev.py:1201]: no worries - maybe we should add that! Although the whole lint check is fast enough that I'm not sure it matters much.

**Review by j-bowhay — APPROVED** [review]

**Pull request merged**